src/utils/serve-functions.js

```diff
@@ -515,21 +515,35 @@ const setupFunctionsBuilder = async ({ config, errorExit, functionsDirectory, lo
 
   log(`${NETLIFYDEVLOG} Function builder ${chalk.yellow(functionBuilder.builderName)} detected${npmScriptString}.`)
 
-  const debouncedBuild = debounce(getBuildFunction({ functionBuilder, log }), DEBOUNCE_WAIT, {
-    leading: true,
-    trailing: true,
-  })
+  const cache = new Map()
+
+  const buildFunction = getBuildFunction({ functionBuilder, log })
+
+  // `memoizedBuild` will avoid consecutive calls to the build function with
+  // the same arguments until the previous invokation has finished running.
+  const memoizedBuild = (path, eventType) => {
```

Comment:
If the user edits the same file multiple times only the first triggers a build?
Do we want to run the last one too?